### DISCUSSION TITLE
The command "poly" does not exist.

### DISCUSSION THREAD

**Discussion by ttamg** [DavidVujic/python-polylith#88] [Q&A] [answered]
Hi @DavidVujic 

Just an installation issue I noticed on my machine (Mac M1) when setting up a new repo for polylith using your plugin.  

I was creating a new repo with `git init` and `poetry init`.  I already have the two plugins installed as per the installation instructions for my previous polylith repos.  In particular the installation commands both suggest that the plugins are already installed.  For example

```bash
> poetry self add poetry-multiproject-plugin  

The following packages are already present in the pyproject.toml and will be skipped:

  • poetry-multiproject-plugin

If you want to update it to the latest compatible version, you can use `poetry self update`.
If you prefer to upgrade it to the latest available version, you can use `poetry self add package@latest`.

Nothing to add.
```

The same goes for the other poetry plugin

So this should all work find right?  

## The issue

Well I was finding that the `poly` command was not registering with `poetry`.  So if I try any of the `poly` commands I get an error

```bash
> poetry poly create ...

The command "poly" does not exist.
```

## The fix

I found that the fix required me to run 

```bash
poetry self update
```

This seemed to fix the issue and now I can create a workspace.  I wanted to flag as this was a bit unexpected and I wasn't quite sure why it happened.  If others have the same issue try `poetry self update` in the new project repo.

Matt

**Comment by DavidVujic** [comment, marked as answer]
I'll convert this into a Github discussion, if others run in to this behavior in Poetry and look for a solution. Thanks @ttamg! 🙏
